### ISSUE TITLE
`jsx-curly-brace-presence`: the fixer merges children separated by a newline

### ISSUE THREAD

**Issue by brainkim** [eslint-stylistic/eslint-stylistic#1279]
### Validations

- [x] Follow the Code of Conduct
- [x] Read the Contributing Guide
- [x] Checked that there isn't already an issue that reports the same bug

### Describe the bug

The fixer for `jsx-curly-brace-presence` unwraps a string expression child even when a sibling text or expression child sits next to it across whitespace-only JSX text. Unwrapping merges the two into one text node, and the newline between them collapses to a space, so the rendered output changes.

With the sole rule `["error", "never"]`:

```jsx
const el = (
	<div>
		{"a"}
		{"b"}
	</div>
);
```

`--fix` gives:

```jsx
const el = (
	<div>
		a
		b
	</div>
);
```

The compiled children, via `tsc --jsx react`:

```
before   createElement("div", null, "a", "b")     renders "ab"
after    createElement("div", null, "a b")        renders "a b"
```

Two children became one, and a space appeared. The whitespace between two expression children is stripped by JSX. The same whitespace between two text children is kept, and its newline collapses to a space.

#### The guard already exists, one case short

The rule does not report this, correctly:

```jsx
const el = <div>{"a"}{"b"}</div>;
```

Zero reports. Unwrapping would join the two into `ab`, and the rule knows it.

So the adjacency check is there. It reads source adjacency, and what matters is adjacency after JSX strips whitespace-only text. A newline between the two children hides them from the check while leaving them adjacent to the compiler.

#### Every shape, before and after

| source | before | after | |
| --- | --- | --- | --- |
| `<div>{"a"}{"b"}</div>` | `"a", "b"` | `"a", "b"` | not reported |
| the two on separate lines | `"a", "b"` | `"a b"` | **changed** |
| `hello` then `{"a"}` on the next line | `"hello", "a"` | `"hello a"` | **changed** |
| `{"a"}` alone in the element | `"a"` | `"a"` | fine |

#### Suggested fix

Extend the existing adjacency check past whitespace-only `JSXText`. A string expression child is safe to unwrap only when no text or expression sibling sits next to it once that whitespace is dropped.

#### Related

#3995 in `eslint-plugin-react` is the same family, a different case: there a `{""}` separator is removed. It reproduces in `@stylistic` 5.10.0 as well.

### Reproduction

Two files, no other setup.

`eslint.config.js`:

```js
import stylistic from "@stylistic/eslint-plugin";
import tsParser from "@typescript-eslint/parser";

export default [{
	files: ["**/*.tsx"],
	languageOptions: {parser: tsParser, parserOptions: {ecmaFeatures: {jsx: true}}},
	plugins: {"@stylistic": stylistic},
	rules: {"@stylistic/jsx-curly-brace-presence": ["error", "never"]},
}];
```

`a.tsx`:

```jsx
export const el = (
	<div>
		{"a"}
		{"b"}
	</div>
);
```

Then `npx eslint --fix a.tsx`, and compare `npx tsc --jsx react --outDir out a.tsx` before and after.

### System Info

```shell
  System:
    OS: macOS 26.5.2
    CPU: (8) arm64 Apple M2
    Memory: 131.27 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 26.7.0
    npm: 11.19.0
    bun: 1.3.14
    Deno: 2.9.2
  npmPackages:
    @stylistic/eslint-plugin: ^5.0.0 => 5.10.0
    @typescript-eslint/parser: ^8.0.0 => 8.67.0
    eslint: ^9.0.0 => 9.39.5
```

Also reproduces on ESLint 10.10.0.

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
